src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/aws/CloudFormationTemplateBuilder.java

```diff
@@ -22,10 +25,13 @@
     @Autowired
     private Configuration freemarkerConfiguration;
 
-    public String build(String templatePath, boolean spotPriced, List<InstanceGroup> templates) {
+    public String build(Stack stack, String templatePath, boolean spotPriced, List<InstanceGroup> templates) {
```

Comment:
If we pass the stack object to this method, it is unnecessary to pass the spotPriced and a List of InstanceGroups as well because they can be queried from the stack. BTW why is the `List<InstanceGroup>` called templates? :)